core-kotlin-modules/core-kotlin-date-time/src/test/kotlin/com/baeldung/timedifference/TimePeriodDifferenceUnitTest.kt

```diff
@@ -0,0 +1,42 @@
+package com.baeldung.timedifference
+
+import org.junit.jupiter.api.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.days
+import kotlin.time.Duration.Companion.hours
+import kotlin.time.Duration.Companion.minutes
+import kotlin.time.Duration.Companion.seconds
+import kotlin.time.TimeSource
+import kotlinx.datetime.Instant
+
+class TimePeriodDifferenceUnitTest {
+
+    @Test
+    fun `given two time marks, when find difference, then give correct time difference`() {
+        val timeSource = TimeSource.Monotonic
+        val mark1 = timeSource.markNow()
+        val mark2 = mark1 + 5.seconds // use timeSource.markNow() to mark current time
+        val difference = mark2 - mark1
+        assertTrue(5.seconds <= difference)
+    }
```

Comment:
Hmm. So you avoided the second `markNow()` to make the test predictable, but now it's a bit hard to understand what is being tested. How about something like this instead:
```suggestion
    @Test
    fun `given two time marks, when find difference, then give correct time difference`() {
        val timeSource = TimeSource.Monotonic
        val mark1 = timeSource.markNow()
        // call operation to measure duration of
        val mark2 = timeSource.markNow()
        val difference = mark2 - mark1
        assertThat(difference).isGreaterThanOrEqualTo(0);
    }
```